File: C++/6thOct.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{   
    int n;cin>>n;
    int c=ceil(sqrt((float)n));
    int arr[c][c];  //answer 2-d array
    int r1 = 0,c1 = 0,r2 = c,c2 = c,num = 1;

	while (r1 < r2 && c1 < c2) {
		for (int i = c1; i < c2; i++) arr[r1][i] = num++;
		r1++;
		for (int i = r1; i < r2; i++) arr[i][c2 - 1] = num++;
		c2--;
		for (int i = c2 - 1; i >= c1; i--) arr[r2 - 1][i] = num++;
		r2--;
		for (int i = r2 - 1; i >= r1; i--) arr[i][c1] = num++;
		c1++;
	}
    for (int i = 0; i < c; i++)
    {
        for (int j = 0; j < c; j++)
        {
            if(arr[i][j]>n)cout<<"  ";
            else
            cout<<arr[i][j]<<" ";   
        }
        cout<<"\n";
    }
    
}

File: JavaScript/6thOct.js
function isPrime(num)
{
     let root=Math.floor(Math.sqrt(num));
     for(var i=2; i <= root;i=i+1)
     {
         if(num%i==0)
         {
             return false
         }
     }
     return true
}

function fib (num,array)
{
    for (let i =2;i<=num;i++)
    {
        array[i]=array[i-1]+array[i-2];
    }
    return array
}
function printer(array)
{
    fibs=new Array()
    fibs.push(1)
    fibs.push(1)
    fibs.push(2)
    for (let v=4; v<array.length;v=v+1) {
        if(isPrime(array[v]) ||  (array[v]%5==0))
        {
            fibs.push(0)
        }
        else 
        {
            fibs.push(array[v])
        }
    }
    return fibs
}
function main()
{
    var n=Number(window.prompt("Enter n: "))
    arr=new Array(n+1)
    arr[0]=0
    arr[1]=1
    array=fib(n,arr)
    // console.log(array)
    console.log(printer(array))
    document.write(printer(array))


}
main()